File: src/main/java/red/tetracube/smarthub/controllers/bot/payloads/DeviceFeatureAction.java
package red.tetracube.smarthub.controllers.bot.payloads;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

public record DeviceFeatureAction(
        @JsonProperty("deviceName") String deviceName,
        @JsonProperty("featureName") String featureName,
        @JsonProperty("commands") List<String> commands
) {
}

File: src/main/kotlin/iot/tetracubered/bot/payloads/DeviceFeatureAction.kt
package iot.tetracubered.bot.payloads

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty

data class DeviceFeatureAction @JsonCreator constructor(
    @field:JsonProperty("deviceName") val deviceName: String,
    @field:JsonProperty("featureName") val featureName: String,
    @field:JsonProperty("commands") val commands: List<String>
)
